### PR TITLE
Add route to connect to cloud

### PR DESCRIPTION
**Pull request by bkeepers**
This adds `/connect/:token` route to configure a Flipper Cloud token and saves it in the session. It uses a preshared key to encrypt and sign the token in transit, so we can set expiry on them and not have to worry as much about replay attacks.

### CODE REVIEW

File: lib/flipper/middleware/session.rb [hunk 1]
```diff
@@ -7,9 +9,29 @@ def initialize(app, env_key: 'flipper')
...
+
+  # Returns a Flipper instance using the given session
+  def self.configure(session, token: session[:cloud_token])
+    adapter = Flipper::Adapters::Session.new(session)
+
+    if token
+      Flipper::Cloud.new(
+        token: token,
+        local_adapter: adapter,
+        sync_secret: "disable polling"
```
Comment (jnunemaker) [inline lib/flipper/middleware/session.rb:31]: Seems like we need a better way to disable polling so this works into the future.

**Review by jnunemaker — APPROVED** [review]
Looks neat. 

So it does a request to cloud before every request currently? Just checking if I understand.

**Comment by bkeepers** [comment]
> So it does a request to cloud before every request currently? Just checking if I understand.

That is correct. If this app starts to get decent traffic we can figure out a different approach, like keeping recently used Cloud instances in a cache and syncing them with polling. But it felt like that was going to get very complicated quickly.

File: lib/flipper/middleware/session.rb (hunk 1 above)
Comment (bkeepers) [reply lib/flipper/middleware/session.rb:31]: Yes, I made a note to self to followup on this.

**Pull request merged**